### PR TITLE
Replace custom `attrdict` with `ml_collections` implementation.

### LINKED ISSUE #1690: Sub custom attrdict for ml collections
https://github.com/google/ml_collections

### PR DESCRIPTION
**Pull request by martinkim0**
Closes #1690.

### CODE REVIEW

**Comment by martinkim0** [comment]
Seems that `ml_collections` does not allow `ConfigDict`s to be initialized with nested dictionaries where inner dictionaries have non-iterable types (e.g. `int`s) as keys ([#13](https://github.com/google/ml_collections/issues/13)). This comes up as a problem when initializing `~scvi.model.TOTALVI`, since `"protein_batch_mask"` has `int`s as keys.

File: docs/api/developer.md [hunk 1]
```diff
@@ -246,5 +246,4 @@ Utility functions used by scvi-tools.
 
    utils.track
    utils.setup_anndata_dsp
-   utils.attrdict
```
Comment (adamgayoso) [inline docs/api/developer.md:249]: Removing the old one completely would be a breaking change. Can we just create a dummy attrdict object that inherits completely an ml collection?

**Review by adamgayoso — CHANGES_REQUESTED** [review]

File: scvi/utils/_attrdict.py [hunk 2]
```diff
@@ -1,40 +1,11 @@
...
-    Parameters
-    ----------
-    recursive
-        If True, recursively converts nested dictionaries into :class:`~scvi.utils.attrdict` objects.
-
-    Notes
-    -----
-    Based off of https://stackoverflow.com/questions/38034377/object-like-attribute-access-for-nested-dictionary.
+    Dummy class to allow for backwards compatibility with old attrdict
+    Now uses ml-collections FrozenConfigDict.
```
Comment (adamgayoso) [inline scvi/utils/_attrdict.py:7]: we should be able to add ml colletions to intersphinx in conf.py and do a proper :class: link to the object.

https://ml-collections.readthedocs.io/en/latest/

File: docs/api/developer.md (hunk 1 above)
Comment (martinkim0) [reply docs/api/developer.md]: Addressed with new changes

**Review by adamgayoso — CHANGES_REQUESTED** [review]
LGTM but can you add the package to pyproject?

**Pull request merged**